proposed/http-client/http-client.md

```diff
@@ -27,6 +27,10 @@ should not follow redirect nor throw exceptions on HTTP responses with status 4x
 The following interfaces MAY be implemented together within a single class or
 in separate classes.
 
+An implementing library MUST implement `Psr\Http\Client\Exception` for each exception it throws. 
+It SHOULD implement exceptions for `Psr\Http\Client\Exception\NetworkException` and
+`Psr\Http\Client\Exception\RequestException`.
```

Comment:
should we give an explanation why it should do that? to allow for the client to more precisely act on errors?

for the flow of this document: the paragraph above says "The following interfaces..." and now we add another one talking about exceptions. could we move this somethwere else? maybe a chapter on exceptions?